### PR TITLE
[SPARK-8930][SQL]Throw a AnalysisException with meaningful messages if DataFrame#explode takes a star in expressions

### PR DESCRIPTION
**Pull request by yjshen**
This PR tries to give a meaningful error message when users called DataFrame#explode method using \* as column name, since the explode's second parameter `f` directly relies on the row structure the first parameter `input: Column*` returns, it's better to give specific wanted columns instead of relying on star unfolding.

JIRA: https://issues.apache.org/jira/browse/SPARK-8930

### CODE REVIEW

**Comment by yjshen** [comment]
cc @rxin

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 1]
```diff
@@ -602,6 +602,8 @@ class Analyzer(
    */
   object ResolveGenerate extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case g: Generate if ResolveReferences.containsStar(g.generator.children) =>
+        failAnalysis("* is not considered as valid explode input, use concrete column instead")
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: "cannot explode *"

**Comment by yjshen** [comment]
@yhuai , can you review the PR please? Thanks!

**Comment by yhuai** [comment]
LGTM. I am merging it to master and branch 1.5.

**Comment by yhuai** [comment]
@yjshen Let's add a summary of your change (including the issue we are addressing in this PR) and a link to the jira in the description part of the PR, which help others to catch up changes quickly.

**Comment by yjshen** [comment]
@yhuai , thanks, I've updated this PR's description first, will follow this in my later PRs.

**Pull request closed without merging**